src/main/java/com/google/sps/servlets/FeedbackServlet.java

```diff
@@ -0,0 +1,96 @@
+// // distributed under the License is distributed on an "AS IS" BASIS,
+// // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// // See the License for the specific language governing permissions and
+// // limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+import com.google.sps.data.Feedback;
+import com.google.sps.data.SearchItem;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonParser;
+import com.google.gson.Gson;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.Collection;
+import com.google.gson.reflect.TypeToken;
+import java.util.Optional;
+
+@WebServlet("/feedback")
+public class FeedbackServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String user = request.getParameter("user");
+        Filter propertyFilter = new FilterPredicate("user", FilterOperator.EQUAL, user);
+        Query query = new Query("Feedback").setFilter(propertyFilter);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Feedback> feedbackList = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            String userID = (String) entity.getProperty("user");
+            String restaurantName = (String) entity.getProperty("restaurantName");
+            String restaurantRating = (String) entity.getProperty("restaurantRating");
+            String rrpRating = (String) entity.getProperty("rrpRating");
+            String notes = (String) entity.getProperty("notes");
+            Feedback feedback = new Feedback(userID, restaurantName, restaurantRating, rrpRating, notes);
+            feedbackList.add(feedback);
+        }
+        // Send the JSON as the response
+        response.setContentType("application/json");
+        response.getWriter().println(new Gson().toJson(feedbackList));
+    }
```

Comment:
new line after this one

Reply:
in the file, there is a new line but it's not recognizing it here (marking it as Outdated)